ophyd/signal.py

```diff
@@ -326,6 +337,8 @@ def set(self, value, *, timeout=None, settle_time=None, **kwargs):
             kwargs,
         )
 
+        settle_time = settle_time or self.settle_time
```

Comment:
This needs to be 
```suggestion
        settle_time = settle_time if settle_time is not None else self.settle_time
```

as a 0 is "falsey" but is a valid settle time and should be respected.